src/app/shared/expiration/date-picker/expiration-date.component.ts

```diff
@@ -91,7 +91,7 @@ export class ExpirationDateComponent implements OnInit, OnDestroy {
     }
   }
 
-  private calcSeconds(value: string): number {
+  private calcSeconds(value: Date): number {
     const d = new Date(value);
```

Comment:
Can this line be removed if the param type is `Date`?
```suggestion
  private calcSeconds(value: Date): number {
```
By the way, is this change related to the angular 14 upgrade?

Reply:
yes, it was complaining on `string` here. 
Checked: it is NOT safe to remove that Date creation. User can pass string to the input (he doesn't need to use callendar). If he pass string, then we do not have Date object, we have string. I will adjust typing that it might be `Date | string`